### ISSUE TITLE
Partial template specialisation in a namespace

### ISSUE THREAD

**Issue by NOhs** [michaeljones/breathe#442]
I have the following C++ file

```cpp
namespace test
{
/*! some doc
 */
template<typename Func, in Dim>
struct Data{};

/*! some doc for partial specialisation
 */
template<typename Func>
struct Data<Func, 1>{};
};
```

In my rst file the following works:

```rst
.. doxygenstruct:: test::Data
```

But the following fails:

```rst
.. doxygenstruct:: test::Data<Func, 1>
```

With the warning:

```
doxygenstruct: Cannot find class “test::Data<Func, 1>” in doxygen xml output for project “test_project” from directory: cpp/xml/
```

Is this type of partial template specialisation supported? Is my call to `doxygenstruct` wrong?

Sphinx: 2.1.1
Breathe: 4.13.0.post0

Thanks btw for this cool project :).

**Comment by melvinvermeeren** [comment]
Could you post the relevant doxygen XML? I am thinking this may be doxygen splitting things up in a certain way with the templates, which then causes breathe's failure to find it.

**Comment by vermeeren** [comment]
This might be fixed with #512, released in Breathe v4.17.0. Note that you also need Sphinx 3.x for recent Breathe versions. Could someone try and post results?
